libraries/utilities/plugin_code_generator.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/python
+
+# Copyright (c) 2018 Blockchain B.V.
+#
+# The MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+#################################################################################
+# This script is called everytime cmake is called. It generates code to certain #
+# files which are related to the creation of plugins.                           #
+# The working directory of the script is the project root directory.            #
+# Script is called from libraries/utilities/CMakeLists.txt                      #
+#################################################################################
+
+
+import json 
+import os
+
+
+# list of files where generated code is used (relative to project root)
+# libraries/app/include/graphene/api.hpp
+# libraries/app/api.cpp
+# libraries/app/application.cpp
+
+generated_files_root_dir = "libraries/utilities/include/graphene/utilities/"
```

Comment:
The generated files are not really headers (except `generated_api_hpp_include_code.hpp` maybe), therefore they shouldn't be put into the `include` directory. Also don't use `.hpp` for them, perhaps `.hxx` or `.cxx` depending on content.

Reply:
Will change that for more clearance.

But generally would you say this is way of solving this issue? I think it looks kinda weird to do includes in the middle of the code. But I heard in some projects this is common. Still wouldn't call it good practice.

Reply:
Well, I wouldn't call it beautiful. :-)
But if it gets the job done it's ok I guess.

Reply:
Yeah, that's okay for me. :)